src/prometheus.rs

```diff
@@ -0,0 +1,54 @@
+// Middleware + prometheus exporter setup
+
+use std::time::Instant;
+
+use axum::{extract::MatchedPath, http::Request, middleware::Next, response::IntoResponse};
+use metrics_exporter_prometheus::{Matcher, PrometheusBuilder, PrometheusHandle};
+
+pub fn setup_metrics_recorder() -> PrometheusHandle {
+    const EXPONENTIAL_SECONDS: &[f64] = &[
+        0.005, 0.01, 0.025, 0.05, 0.1, 0.25, 0.5, 1.0, 2.5, 5.0, 10.0,
```

Comment:
nit: we might want to go a bit higher, as our ingress timeout is 60s, and we sometimes reach it when MSK is broken